backend/models/riskMitigation.js

```diff
@@ -0,0 +1,156 @@
+const db = require("../handler/db");
+const responseCodes = require("../response_codes.js");
+const TeamspaceSettings = require("./teamspaceSetting");
+
+const fieldTypes = {
+	"associated_activity": "[object String]",
+	"category": "[object String]",
+	"element": "[object String]",
+	"location_desc": "[object String]",
+	"mitigation_desc": "[object String]",
+	"mitigation_detail": "[object String]",
+	"mitigation_stage": "[object String]",
+	"mitigation_type": "[object String]",
+	"risk_factor": "[object String]",
+	"scope": "[object String]"
+};
+
+class RiskMitigation {
+	getRiskMitigationCollection(account) {
+		return db.getCollection(account, "mitigations");
+	}
+
+	async clearAll(account) {
+		const mitigationColl = await this.getRiskMitigationCollection(account);
+		return await mitigationColl.remove({});
+	}
+
+	async getCriteria(account) {
+		const mitigationColl = await this.getRiskMitigationCollection(account);
+		const attributeBlacklist = ["mitigation_desc", "mitigation_detail"];
+		const criteriaFields = Object.keys(_.omit(fieldTypes, attributeBlacklist));
+		const criteriaPromises = [];
+		const criteria = {};
+
+		// Get teamspace categories
+		let teamspaceCategories = await TeamspaceSettings.getRiskCategories(account);
+		teamspaceCategories = teamspaceCategories.map(x => x.label);
+
+		criteriaFields.forEach((field) => {
+			const promise = mitigationColl.distinct(field, {});
+			criteriaPromises.push(promise);
+			promise.then((values) => {
+				criteria[field] = values;
+			});
+		});
+
+		return Promise.all(criteriaPromises).then(() => {
+			// Append teamspace categories
+			criteria["category"] = [...new Set(Object.assign([], criteria.category).concat(teamspaceCategories))];
+
+			return criteria;
+		});
+	}
+
+	async findMitigationSuggestions(account, criteria) {
+		const mitigationColl = await this.getRiskMitigationCollection(account);
+		const attributeBlacklist = [
+			"mitigation_desc",
+			"mitigation_detail",
+			"mitigation_stage",
+			"mitigation_type"
+		];
+		const criteriaFilterFields = Object.keys(_.omit(fieldTypes, attributeBlacklist));
+
+		// Only pick supported fields and clean empty criteria
+		criteria = _.pick(criteria, criteriaFilterFields);
+		Object.keys(criteria).forEach((key) => {
+			if (criteria[key] === null || criteria[key] === "") {
+				delete criteria[key];
+			}
+		});
+
+		const suggestions = await mitigationColl.find(criteria).toArray();
+
+		return suggestions;
+	}
+
+	async importCSV(account, data) {
+		const csvFields = [
+			"mitigation_desc",
+			"mitigation_detail",
+			"mitigation_stage",
+			"mitigation_type",
+			"category",
+			"location_desc",
+			"element",
+			"risk_factor",
+			"scope",
+			"associated_activity"
+		];
+
+		// remove column headers defined in template
+		data = data.replace(/Treatment Title,Treatment Details,Treatment Stage,Treatment Type,Risk Category,Risk Location,Element Type,Risk Factor,Construction Scope,Associated Activity\r\n/gm,"");
```

Comment:
My test case has the header on multiple rows and also I thought some people might not upload a file with a header. Maybe I'm thinking too much.

Reply:
I think we should do one of 2 ways as discussed in the call, either:

1. take the first line, check the header is identical to what you expect, if not, return a 400
2. just ignore the first line and assume that's the header

1 is probably a better way to do it but requires more work. bottom line is we want to avoid doing a regex on a potentially large file